sqlglot/tokens.py

```diff
@@ -947,9 +947,12 @@ def _scan_number(self) -> None:
             elif self._peek.isidentifier():  # type: ignore
                 number_text = self._text
                 literal = []
-                while self._peek.isidentifier():  # type: ignore
-                    literal.append(self._peek.upper())  # type: ignore
+
+                char = self._peek.strip()  # type: ignore
```

Comment:
why do you need to strip

Reply:
Because we want to stop tokenizing a possible VAR if we encounter whitespace.

Reply:
can you do 

while char.strip() and …

Reply:
Done.